src/Compilers/Core/Portable/InternalUtilities/FourState.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents an optional bool? as a single byte.
+    /// </summary>
+    internal enum FourState : byte
```

Comment:
When you get to 4 different states, it's probably time to break it into a named enum.